### PR TITLE
RavenDB-15678  Server-Wide External Replication - Studio

### PR DESCRIPTION
**Pull request by Danielle9897**

### CODE REVIEW

File: src/Raven.Studio/typescript/models/database/tasks/serverWide/serverWideExternalReplicationListModel.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    initializeObservables() {
+        super.initializeObservables();
+
+        this.editUrl = ko.pureComputed(() => appUrl.forEditServerWideExternalReplication(this.taskName()));
+    }
+    
+    // dto param is union-type only so that it compiles - due to class inheritance....
+    update(dto: Raven.Server.Web.System.AdminStudioServerWideHandler.ServerWideTasksResult.ServerWideTask |
+                Raven.Server.Web.System.AdminStudioServerWideHandler.ServerWideTasksResult.ServerWideExternalReplicationTask) {
+        super.update({
```
Comment (ml054) [inline src/Raven.Studio/typescript/models/database/tasks/serverWide/serverWideExternalReplicationListModel.ts]: can't we just call super.update(dto) ?

File: src/Raven.Studio/typescript/viewmodels/database/settings/connectionStrings.ts [hunk 2]
```diff
@@ -145,17 +145,24 @@ class connectionStrings extends viewModelBase {
         return new getConnectionStringsCommand(this.activeDatabase())
             .execute()
             .done((result: Raven.Client.Documents.Operations.ConnectionStrings.GetConnectionStringsResult) => {
+                // ravenEtl
                 this.ravenEtlConnectionStringsNames(Object.keys(result.RavenConnectionStrings));
+                const groupedRavenEtlNames = _.groupBy(this.ravenEtlConnectionStringsNames(), x => this.hasServerWidePrefix(x));
+                const serverWideNames = _.sortBy(groupedRavenEtlNames.true, x => x.toUpperCase());
+                const regularNames = _.sortBy(groupedRavenEtlNames.false, x => x.toUpperCase());
+                this.ravenEtlConnectionStringsNames(regularNames);
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/database/settings/connectionStrings.ts]: this.ravenEtlConnectionStringsNames([...regularNames, ...serverWideNames]);

Comment (ml054) [reply src/Raven.Studio/typescript/viewmodels/database/settings/connectionStrings.ts]: by doing that we notify subscribes once

File: src/Raven.Studio/typescript/viewmodels/manage/createServerWideTask.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import eventsCollector = require("common/eventsCollector");
+import appUrl = require("common/appUrl");
+import router = require("plugins/router");
+import dialogViewModelBase = require("viewmodels/dialogViewModelBase"); 
+
+class createServerWideTask extends dialogViewModelBase {
+
+    compositionComplete() {
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/manage/createServerWideTask.ts]: remove?

File: src/Raven.Studio/typescript/viewmodels/manage/createServerWideTask.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        super.compositionComplete();
+    }
+
+    newServerWideReplicationTask() {
+        eventsCollector.default.reportEvent("ServerWideExternalReplication", "new");
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/manage/createServerWideTask.ts]: IIRC we use camelCase for event names?

File: src/Raven.Studio/wwwroot/App/views/manage/serverWideTasks.html [hunk 5]
```diff
@@ -1,25 +1,88 @@
...
+                        <h5 class="tasks-list-item external-replication no-text-transform"><i class="icon-external-replication"></i>
+                            <span>Server-Wide External Replication</span>
+                            <span data-bind="text: ' (' + serverWideExternalReplicationTasks().length + ')'"></span>
+                        </h5><hr>
+                    </div>
+                    <div data-bind="foreach: serverWideExternalReplicationTasks">
+
+                        <div class="panel destination-item external-replication">
+                            <div data-bind="attr: { 'data-state-text': badgeText, class: 'state ' + badgeClass() }">
+                                <!-- todo - replace icon with icon-server-wide-external-replication -->
```
Comment (ml054) [inline src/Raven.Studio/wwwroot/App/views/manage/serverWideTasks.html]: did you report issue for Rafal?

File: src/Raven.Studio/typescript/viewmodels/manage/createServerWideTask.ts (hunk 3 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/manage/createServerWideTask.ts]: removed

File: src/Raven.Studio/typescript/viewmodels/manage/createServerWideTask.ts (hunk 4 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/manage/createServerWideTask.ts]: done

File: src/Raven.Studio/typescript/viewmodels/database/settings/connectionStrings.ts (hunk 2 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/database/settings/connectionStrings.ts]: done

File: src/Raven.Studio/typescript/models/database/tasks/serverWide/serverWideExternalReplicationListModel.ts (hunk 1 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/models/database/tasks/serverWide/serverWideExternalReplicationListModel.ts]: can, done

File: src/Raven.Studio/wwwroot/App/views/manage/serverWideTasks.html (hunk 5 above)
Comment (Danielle9897) [reply src/Raven.Studio/wwwroot/App/views/manage/serverWideTasks.html]: Will open all related issues shortly

**Review by ml054 — APPROVED** [review]

**Pull request merged**